### ISSUE TITLE
protokube / etcd-events failure kops 1.8.11-1.9.0

### ISSUE THREAD

**Issue by henrydobson** [kubernetes/kops#5121]
1. What `kops` version are you running? The command `kops version`, will display
 this information.

```
Version 1.8.1 (git-94ef202) 
# also attempted on  
Version 1.9.0
```

2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.

```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.1", GitCommit:"d4ab47518836c750f9949b9e0d387f20fb92260b", GitTreeState:"clean", BuildDate:"2018-04-13T22:29:03Z", GoVersion:"go1.9.5", Compiler:"gc", Platform:"darwin/amd64"}

Server Version specified 1.8.11
```

3. What cloud provider are you using?

```
aws
```

4. What commands did you run?  What is the simplest way to reproduce this issue?

```
kops create cluster --name=<redacted> \
  --state="s3://<redacted>/kops-state" \
  --authorization=AlwaysAllow \
  --zones=eu-central-1a,eu-central-1b,eu-central-1c \
  --master-zones=eu-central-1a,eu-central-1b,eu-central-1c \
  --node-count=1 \
  --node-size=r4.large \
  --master-count=3 \
  --master-size=t2.medium \
  --kubernetes-version=1.8.11 \
  --bastion="true" \
  --topology=private \
  --network-cidr=<redacted>/20 \
  --networking=calico \
  --out=/tmp/kops-out \
  --target=terraform
```

5. What happened after the commands executed?

```
New cluster is created successfully and publishes cluster assets to s3.
I then apply terraform.
Wait for the cluster to become ready.

###

kops validate cluster ...
... omitted 

Component Failures
NAME
etcd-1

... omitted

###

kubectl get cs
NAME                 STATUS      MESSAGE                                                                                     ERROR
etcd-1               Unhealthy   Get http://127.0.0.1:4002/health: dial tcp 127.0.0.1:4002: getsockopt: connection refused
scheduler            Healthy     ok
controller-manager   Healthy     ok
etcd-0               Healthy     {"health": "true"}

###

journalctl -b | grep etcd

kubelet[1763]: I0509 10:56:34.099073    1763 kuberuntime_manager.go:503] Container {Name:kube-apiserver Image:gcr.io/google_containers/kube-apiserver:v1.8.11 Command:[/bin/sh -c /usr/local/bin/kube-apiserver --address=127.0.0.1 --admission-control=Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,Priority,ResourceQuota --allow-privileged=true --anonymous-auth=false --apiserver-count=3 --authorization-mode=AlwaysAllow --basic-auth-file=/srv/kubernetes/basic_auth.csv --client-ca-file=/srv/kubernetes/ca.crt --cloud-provider=aws --etcd-servers-overrides=/events#http://127.0.0.1:4002 --etcd-servers=http://127.0.0.1:4001 --insecure-port=8080 --kubelet-preferred-address-types=InternalIP,Hostname,ExternalIP --proxy-client-cert-file=/srv/kubernetes/apiserver-aggregator.cert --proxy-client-key-file=/srv/kubernetes/apiserver-aggregator.key --requestheader-allowed-names=aggregator --requestheader-client-ca-file=/srv/kubernetes/apiserver-aggregator-ca.cert --requestheader-extra-headers-prefix=X-Remote-Extra- --requestheader-group-headers=X-Remote-Group --requestheader-username-headers=X-Remote-User --secure-port=443 --service-cluster-ip-range=100.64.0.0/13 --storage-backend=etcd2 --tls-cert-file=/srv/kubernetes/server.cert --tls-private-key-file=/srv/kubernetes/server.key --token-auth-file=/srv/kubernetes/known_tokens.csv --v=2 2>&1 | /bin/tee -a /var/log/kube-apiserver.log] Args:[] WorkingDir: Ports:[{Name:https HostPort:443 ContainerPort:443 Protocol:TCP HostIP:} {Name:local HostPort:8080 ContainerPort:8080 Protocol:TCP HostIP:}] EnvFrom:[] Env:[] Resources:{Limits:map[] Requests:map[cpu:{i:{value:150 scale:-3} d:{Dec:<nil>} s:150m Format:DecimalSI}]} VolumeMounts:[{Name:etcssl ReadOnly:true MountPath:/etc/ssl SubPath: MountPropagation:<nil>} {Name:etcpkitls ReadOnly:true MountPath:/etc/pki/tls SubPath: MountPropagation:<nil>} {Name:etcpkica-trust ReadOnly:true MountPath:/etc/pki/ca-trust SubPath: MountPropagation:<nil>} {Name:usrsharessl ReadOnly:
kubelet[1763]: I0509 10:56:34.128261    1763 kube_docker_client.go:333] Stop pulling image "gcr.io/google_containers/etcd:2.2.1": "Status: Downloaded newer image for gcr.io/google_containers/etcd:2.2.1"

kubelet[1763]: I0509 10:56:34.795452    1763 kubelet.go:1871] SyncLoop (PLEG): "etcd-server-redacted.eu-central-1.compute.internal_kube-system(7cee2c08938b70809406f4f7d61b2e2d)", event: &pleg.PodLifecycleEvent{ID:"7cee2c08938b70809406f4f7d61b2e2d", Type:"ContainerStarted", Data:"625b32a946802763cbbb09e62e2b3404ca5b99523f89523fb013f982bbc9fb42"}

kubelet[1763]: W0509 10:56:34.812856    1763 status_manager.go:431] Failed to get status for pod "etcd-server-redacted.eu-central-1.compute.internal_kube-system(7cee2c08938b70809406f4f7d61b2e2d)": Get https://127.0.0.1/api/v1/namespaces/kube-system/pods/etcd-server-redacted-eu-central-1.compute.internal: dial tcp 127.0.0.1:443: getsockopt: connection refused

kubelet[1763]: E0509 10:56:34.813410    1763 kubelet.go:1612] Failed creating a mirror pod for "etcd-server-redacted.eu-central-1.compute.internal_kube-system(7cee2c08938b70809406f4f7d61b2e2d)": Post https://127.0.0.1/api/v1/namespaces/kube-system/pods: dial tcp 127.0.0.1:443: getsockopt: connection refused

kubelet[1763]: E0509 10:56:35.819486    1763 kubelet.go:1612] Failed creating a mirror pod for "etcd-server-redacted.eu-central-1.compute.internal_kube-system(7cee2c08938b70809406f4f7d61b2e2d)": Post https://127.0.0.1/api/v1/namespaces/kube-system/pods: dial tcp 127.0.0.1:443: getsockopt: connection refused

kubelet[1763]: E0509 10:57:55.774715    1763 event.go:200] Server rejected event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"ip-redacted.eu-central-1.compute.internal.152cf4a8975f4206", GenerateName:"", Namespace:"default", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, InvolvedObject:v1.ObjectReference{Kind:"Node", Namespace:"", Name:"ip-redacted.compute.internal", UID:"ip-redacted.eu-central-1.compute.internal", APIVersion:"", ResourceVersion:"", FieldPath:""}, Reason:"NodeHasSufficientMemory", Message:"Node ip-redacted.eu-central-1.compute.internal status is now: NodeHasSufficientMemory", Source:v1.EventSource{Component:"kubelet", Host:"ip-redacted.eu-central-1.compute.internal"}, FirstTimestamp:v1.Time{Time:time.Time{sec:63661460178, nsec:703041030, loc:(*time.Location)(0x537fac0)}}, LastTimestamp:v1.Time{Time:time.Time{sec:63661460181, nsec:853278133, loc:(*time.Location)(0x537fac0)}}, Count:6, Type:"Normal"}': 'client: etcd cluster is unavailable or misconfigured; error #0: dial tcp 127.0.0.1:4002: getsockopt: connection refused

docker[1668]: I0509 11:16:37.873212    1723 volume_mounter.go:250] Skipping mount of master volume "vol-redacted", because etcd cluster "main" is already mounted

###

TL;DR
components are NOT healthy
etcd mounts have been skipped by protokube because protokube thinks they are already mounted but they are not. 
etcd-events service is not running on 4002 
```

6. What did you expect to happen?

```
All components should be healthy
protokube should mount all etcd volumes
etcd-events should be healthy on 4002
```

7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -oyaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.

```
apiVersion: kops/v1alpha2
kind: Cluster
metadata:
  creationTimestamp: 2018-05-09T10:54:11Z
  name: redacted
spec:
  api:
    loadBalancer:
      type: Public
  authorization:
    alwaysAllow: {}
  channel: stable
  cloudProvider: aws
  configBase: s3://redacted
  etcdClusters:
  - etcdMembers:
    - instanceGroup: master-eu-central-1a
      name: a
    - instanceGroup: master-eu-central-1b
      name: b
    - instanceGroup: master-eu-central-1c
      name: c
    name: main
  - etcdMembers:
    - instanceGroup: master-eu-central-1a
      name: a
    - instanceGroup: master-eu-central-1b
      name: b
    - instanceGroup: master-eu-central-1c
      name: c
    name: events
  iam:
    allowContainerRegistry: true
    legacy: false
  kubernetesApiAccess:
  - 0.0.0.0/0
  kubernetesVersion: 1.8.11
  masterPublicName: api.redacted
  networkCIDR: 10.2.144.0/20
  networking:
    calico: {}
  nonMasqueradeCIDR: 100.64.0.0/10
  sshAccess:
  - 0.0.0.0/0
  subnets:
  - cidr: 10.2.146.0/23
    name: eu-central-1a
    type: Private
    zone: eu-central-1a
  - cidr: 10.2.148.0/23
    name: eu-central-1b
    type: Private
    zone: eu-central-1b
  - cidr: 10.2.150.0/23
    name: eu-central-1c
    type: Private
    zone: eu-central-1c
  - cidr: 10.2.144.0/26
    name: utility-eu-central-1a
    type: Utility
    zone: eu-central-1a
  - cidr: 10.2.144.64/26
    name: utility-eu-central-1b
    type: Utility
    zone: eu-central-1b
  - cidr: 10.2.144.128/26
    name: utility-eu-central-1c
    type: Utility
    zone: eu-central-1c
  topology:
    bastion:
      bastionPublicName: bastion.redacted
    dns:
      type: Public
    masters: private
    nodes: private

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-05-09T10:54:11Z
  labels:
    kops.k8s.io/cluster: redacted
  name: bastions
spec:
  image: kope.io/k8s-1.8-debian-jessie-amd64-hvm-ebs-2018-02-08
  machineType: t2.micro
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: bastions
  role: Bastion
  subnets:
  - utility-eu-central-1a
  - utility-eu-central-1b
  - utility-eu-central-1c

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-05-09T10:54:11Z
  labels:
    kops.k8s.io/cluster: redacted 
  name: master-eu-central-1a
spec:
  image: kope.io/k8s-1.8-debian-jessie-amd64-hvm-ebs-2018-02-08
  machineType: t2.medium
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-eu-central-1a
  role: Master
  subnets:
  - eu-central-1a

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-05-09T10:54:11Z
  labels:
    kops.k8s.io/cluster: redacted
  name: master-eu-central-1b
spec:
  image: kope.io/k8s-1.8-debian-jessie-amd64-hvm-ebs-2018-02-08
  machineType: t2.medium
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-eu-central-1b
  role: Master
  subnets:
  - eu-central-1b

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-05-09T10:54:11Z
  labels:
    kops.k8s.io/cluster: redacted
  name: master-eu-central-1c
spec:
  image: kope.io/k8s-1.8-debian-jessie-amd64-hvm-ebs-2018-02-08
  machineType: t2.medium
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-eu-central-1c
  role: Master
  subnets:
  - eu-central-1c

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-05-09T10:54:11Z
  labels:
    kops.k8s.io/cluster: redacted
  name: nodes
spec:
  image: kope.io/k8s-1.8-debian-jessie-amd64-hvm-ebs-2018-02-08
  machineType: r4.large
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: nodes
  role: Node
  subnets:
  - eu-central-1a
  - eu-central-1b
  - eu-central-1c
```
8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.

9. Anything else do we need to know?

**Comment by henrydobson** [comment]
Misconfigured etcd ebs volume tags
